core/templates/pages/skill-editor-page/editor-tab/skill-misconceptions-editor/misconception-editor.component.spec.ts

```diff
@@ -0,0 +1,212 @@
+
+    expect(component.notesEditorIsOpen).toBe(true);
+  });
+
+  it('should open feedback editor when clicking on edit button', () => {
+    expect(component.feedbackEditorIsOpen).toBe(false);
+
+    component.openFeedbackEditor();
+
+    expect(component.feedbackEditorIsOpen).toBe(true);
+  });
+
+  it('should save name when clicking on save button', () => {
+    let updateNameSpy = spyOn(
+      skillUpdateService, 'updateMisconceptionName').and.returnValue(null);
+    component.openNameEditor();
+    // Setting new name.
+    component.container.misconceptionName = 'newName';
+
+    component.saveName();
+
+    expect(updateNameSpy).toHaveBeenCalled();
+  });
+
+  it('should save notes when clicking on save button', () => {
+    let updateNotesSpy = spyOn(
+      skillUpdateService, 'updateMisconceptionNotes').and.returnValue(null);
+    component.openNotesEditor();
+    // Setting new notes content.
+    component.container.misconceptionNotes = 'newNotes';
+
+    component.saveNotes();
+
+    expect(updateNotesSpy).toHaveBeenCalled();
+  });
+
+  it('should save feedback when clicking on save button', () => {
+    let updateFeedbackSpy = spyOn(
+      skillUpdateService, 'updateMisconceptionFeedback').and.returnValue(null);
+    component.openFeedbackEditor();
+    // Setting new feedback content.
+    component.container.misconceptionFeedback = 'newFeedback';
+
+    component.saveFeedback();
+
+    expect(updateFeedbackSpy).toHaveBeenCalledWith(
+      sampleSkill, 1, 'feedback', 'newFeedback');
+  });
+
+  it('should close name editor when clicking on cancel button', () => {
+    component.openNameEditor();
+
+    component.cancelEditName();
+
+    expect(component.nameEditorIsOpen).toBe(false);
+  });
+
+  it('should close notes editor when clicking on cancel button', () => {
+    component.openNotesEditor();
+
+    component.cancelEditNotes();
+
+    expect(component.notesEditorIsOpen).toBe(false);
+  });
+
+  it('should close feedback editor when clicking on cancel button', () => {
+    component.openFeedbackEditor();
+
+    component.cancelEditFeedback();
+
+    expect(component.feedbackEditorIsOpen).toBe(false);
+  });
+
+  it('should address the misconception\'s updates', () => {
+    let updatesSpy = spyOn(
+      skillUpdateService, 'updateMisconceptionMustBeAddressed')
+      .and.returnValue(null);
+
+    component.updateMustBeAddressed();
+
+    expect(updatesSpy).toHaveBeenCalledWith(
+      sampleSkill, 1, true, false);
+  });
+
+  it('should get schema for editing notes', () => {
+    expect(component.getSchemaNotes())
+      .toEqual(component.NOTES_FORM_SCHEMA);
+  });
+
+  it('should get schema for feedback', () => {
+    expect(component.getSchemaFeedback())
+      .toEqual(component.FEEDBACK_FORM_SCHEMA);
+  });
+
+  it('should update misconceptionNotes', () => {
+    let notes = 'notes1';
+    component.updateLocalNotes(notes);
+
+    expect(component.container.misconceptionNotes).toEqual(notes);
+  });
```

Comment:
Can you assign `misconceptionNotes` a value and check if it changes after calling `updateLocalNotes`

Reply:
Done.